### DISCUSSION TITLE
如果要实现openai的api completions中的tools功能应该如何做呢

### DISCUSSION THREAD

**Discussion by keymanlu** [zhayujie/chatgpt-on-wechat#2123] [Q&A]
如果我想在对话过程中，让gpt识别问题中的内容调用一些本地函数，如果原生的openai的api可以用类似
OpenAI.chat.completions.create(...  , tools=xxx .... )
这个tools可以让其调用本地函数，我如果想在这chatgpt-on-wechat项目中实现类似的功能，应该如何做呢？谢谢

**Comment by mengyuxin** [comment]
嗯。我也想加入一个 天气和股票，以及彩票和新闻的函数！
